tests/testAstrometry.py

```diff
@@ -620,33 +675,150 @@ def test_appGeoFromICRS(self):
 
-            ra_test, dec_test = _appGeoFromICRS(numpy.array([ra_icrs]), numpy.array([dec_icrs]),
+            ra_test, dec_test = _appGeoFromICRS(ra_icrs, dec_icrs,
                                                 mjd=obs.mjd,
-                                                pm_ra=numpy.array([pm_ra]),
-                                                pm_dec=numpy.array([pm_dec]),
-                                                v_rad=numpy.array([v_rad]),
-                                                parallax=numpy.array([px]),
+                                                pm_ra=pm_ra,
+                                                pm_dec=pm_dec,
+                                                v_rad=v_rad,
+                                                parallax=px,
                                                 epoch=2000.0)
 
-            distance = arcsecFromRadians(haversine(ra_app, dec_app, ra_test[0], dec_test[0]))
+            self.assertIsInstance(ra_test, np.float)
+            self.assertIsInstance(dec_test, np.float)
+
+            distance = arcsecFromRadians(haversine(ra_app, dec_app, ra_test, dec_test))
             self.assertLess(distance, 0.1)
 
+    def test_appGeoFromICRS_inputs(self):
+        """
+        Test that appGeoFromICRS behaves as expected when given numpy array inputs
+        """
+
+        np.random.seed(83)
+        nSamples = 100
+        ra_icrs = 2.0*np.pi*np.random.random_sample(nSamples)
+        dec_icrs = (np.random.random_sample(nSamples)-0.5)*np.pi
+        pm_ra = radiansFromArcsec((np.random.random_sample(nSamples)-0.5)*0.02)
+        pm_dec = radiansFromArcsec((np.random.random_sample(nSamples)-0.5)*0.02)
+        parallax = radiansFromArcsec(np.random.random_sample(nSamples)*0.01)
+        v_rad = (np.random.random_sample(nSamples)-0.5)*1200.0
+        mjd = ModifiedJulianDate(TAI=59580.0)
+
+        ra_control, dec_control = _appGeoFromICRS(ra_icrs, dec_icrs,
+                                                  pm_ra=pm_ra, pm_dec=pm_dec, parallax=parallax,
+                                                  v_rad=v_rad, mjd=mjd)
+
+        self.assertIsInstance(ra_control, np.ndarray)
+        self.assertIsInstance(dec_control, np.ndarray)
+
+        # test that passing in floats and numpy arrays gives the same result
+        for ix in range(len(ra_control)):
+            ra_test, dec_test = _appGeoFromICRS(ra_icrs[ix], dec_icrs[ix],
+                                                pm_ra=pm_ra[ix], pm_dec=pm_dec[ix],
+                                                parallax=parallax[ix], v_rad=v_rad[ix],
+                                                mjd=mjd)
+
+            self.assertIsInstance(ra_test, np.float)
+            self.assertIsInstance(dec_test, np.float)
+            dd = arcsecFromRadians(haversine(ra_test, dec_test, ra_control[ix], dec_control[ix]))
+            self.assertLess(dd, 1.0e-6)
+
+        # next test that inputs of inappropriate types raise RuntimeErrors
```

Comment:
Is this not mainly testing _validate_inputs rather than _appGeoFromICRS and maybe should be a more general test method?